### PR TITLE
Issue 23212 - improve rt/lifetime.d docs

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

**Review by MrcSnm — APPROVED** [review]
Really liked how it feels much clearer on how to trigger the functions from runtime. The examples, feels much more natural to read. Has a summary on what complex functions does. The addition on the private makes it clearer to check on what it is really necessary on the runtime implementation :D

File: src/rt/lifetime.d [hunk 1]
```diff
@@ -726,12 +754,21 @@ void __doPostblit(void *ptr, size_t len, const TypeInfo ti)
...
+If the array capacity isn't currently large enough
+to hold the requested capacity (in number of elements), then the array is
+resized/reallocated to the appropriate size.
+
+Pass in a requested capacity of 0 to get the current capacity.
+
+Params:
+    ti = type info of element type
+    newcapacity = requested new capacity
+    p = pointer to array to set
```
Comment (rainers) [inline src/rt/lifetime.d:768]: Maybe add the comment about the length mismatch here, too.

File: src/rt/lifetime.d [hunk 2]
```diff
@@ -1444,8 +1526,28 @@ extern (C) void rt_finalizeFromGC(void* p, size_t size, uint attr) nothrow
...
+{
+    int[] a = [1, 2];
+    a.length = 3; // gets lowered to `_d_arraysetlengthT(typeid(a), 3, &a)`
+}
+---
+
+Params:
+    ti = `TypeInfo` of array
+    newlength = new value for the array's `.length`
+    p = pointer to array to update the `.length` of
```
Comment (rainers) [inline src/rt/lifetime.d:1548]: Maybe add the comment about the length mismatch here, too.

File: src/rt/lifetime.d [hunk 3]
```diff
@@ -1444,8 +1526,28 @@ extern (C) void rt_finalizeFromGC(void* p, size_t size, uint attr) nothrow
...
 /**
- * Resize dynamic arrays with 0 initializers.
- */
+Resize a dynamic array by setting the `.length` property
+
+Newly created elements are initialized to their default value.
+
+Has two variants:
+- `_d_arraysetlengthT` for arrays with elements that initialize to 0
+- `_d_arraysetlengthiT` for non-zero initializers from retrieved from `TypeInfo`
```
Comment (rainers) [inline src/rt/lifetime.d:1535]: duplicate "from".

**Review by rainers — APPROVED** [review]
LGTM

File: src/rt/lifetime.d (hunk 1 above)
Comment (dkorpel) [reply src/rt/lifetime.d:768]: The length doesn't matter here

File: src/rt/lifetime.d (hunk 2 above)
Comment (dkorpel) [reply src/rt/lifetime.d:1548]: Here it does

File: src/rt/lifetime.d (hunk 1 above)
Comment (dkorpel) [reply src/rt/lifetime.d]: I added a comment mentioning that

**Pull request merged**